rasterio/merge.py

```diff
@@ -242,6 +255,11 @@ def copyto(old_data, new_data, old_nodata, new_nodata):
             region_nodata = region == nodataval
             temp_nodata = temp.mask
 
-        copyto(region, temp, region_nodata, temp_nodata)
+        sig = signature(copyto)
+
+        if len(sig.parameters.keys()) == 7:
+            copyto(region, temp, region_nodata, temp_nodata, idx, roff, coff)
+        else:
+            copyto(region, temp, region_nodata, temp_nodata)
```

Comment:
@normanb If you change all the builtin copyto functions to `def copyto(old_data, new_data, old_nodata, new_nodata, **kwargs):` then we won't need to test the number of parameters and can drop the dependence on funcsigs and change these lines to

```suggestion
        copyto(region, temp, region_nodata, temp_nodata, index=dx, roff=roff, coff=coff)
```